numba/dppl/ocl/oclimpl.py

```diff
@@ -250,20 +353,16 @@ def _generic_array(context, builder, shape, dtype, symbol_name, addrspace):
         if dtype not in types.number_domain:
             raise TypeError("unsupported type: %s" % dtype)
 
-        # Convert to generic address-space
-        dataptr = context.addrspacecast(builder, gvmem,
-                                        target.SPIR_GENERIC_ADDRSPACE)
-
     else:
         raise NotImplementedError("addrspace {addrspace}".format(**locals()))
 
-    return _make_array(context, builder, dataptr, dtype, shape)
+    return _make_array(context, builder, gvmem, dtype, shape, addrspace=addrspace)
```

Comment:
I know why the change was done, since you explained them to me 👍. But, please add a comment for others.

Reply:
Comment added.